libs/authjs-client/src/lib/gfw-client.ts

```diff
@@ -136,74 +21,119 @@ export class GFW {
+    return await fetch(`${gatewayUrl}/${GFW_API_AUTH_ENDPOINT}/token?access-token=${accessToken}`, {
+      method: 'GET',
+      headers: GFW_CLIENT_BASE_HEADERS,
+    }).then(handleResponseTextOrJson<Token>)
   }
 
   /**
    * Performs logout against the GFW API with the given refresh_token
    * @param gatewayUrl GFW Api gateway url
-   * @param accessToken accessToken to use for logging in
+   * @param refreshToken refreshToken to use for logging in
    * @returns Promise resolving an empty object
    */
-  static async logout(gatewayUrl: string, refreshToken: string): Promise<any> {
-    const authUrl = getAuthUrl(gatewayUrl)
-    const requestData = {
-      url: `${authUrl}/logout`,
+  static async logout(gatewayUrl: string, refreshToken: string): Promise<unknown> {
+    return await fetch(`${gatewayUrl}/${GFW_API_AUTH_ENDPOINT}/logout`, {
       method: 'GET',
+      headers: Object.assign({}, GFW_CLIENT_BASE_HEADERS, {
+        'refresh-token': refreshToken,
+      }),
+    }).then(handleResponseTextOrJson)
+  }
+
+  /**
+   * Performs login against the GFW API with the given access_token
+   * and stores the bearer and refresh tokens in the client config
+   * to be used in the upcoming requests
+   * @returns Promise resolving to the tokens returned from the API.
+   */
+  async login(): Promise<Token> {
+    if (this.config.debug) {
+      console.log('GFW-API. Logging in with access token')
     }
-    const headers = {
-      'refresh-token': refreshToken,
+    try {
+      const results = await GFW.login(this.url, this.config.accessToken ?? '')
+      if (results.token) this.config.bearerToken = results.token
+      if (results.refreshToken) this.config.refreshToken = results.refreshToken
+      return results
+    } catch (e) {
+      if (this.config.debug) {
+        console.error('GFW-API. Error logging in', e)
+      }
+      throw e
     }
-    console.log('GFW-API. Logout')
-    const results = await fetch(requestData.url, {
-      method: requestData.method,
-      headers: Object.assign({}, baseHeaders, headers),
-    }).then(_handleResponseTextOrJson)
+  }
 
-    return results
+  async fetchUser() {
+    try {
+      return await this.get<UserData>(`${GFW_API_AUTH_ENDPOINT}/me`)
+    } catch (e) {
+      if (this.config.debug) {
+        console.log('GFW-API. Error fetching user', e)
+      }
+      throw e
+    }
+  }
+
+  async fetchGuestUser(): Promise<UserData> {
+    try {
+      const permissions: UserPermission[] = await this.get<APIPagination<UserPermission>>(
+        `${GFW_API_AUTH_ENDPOINT}/acl/permissions/anonymous`
+      ).then((response) => {
+        return response.entries
+      })
+      const user: UserData = { id: 0, type: GUEST_USER_TYPE, permissions, groups: [] }
+
+      return user
+    } catch (e) {
+      if (this.config.debug) {
+        console.log('GFW-API. Error fetching guest user', e)
+      }
+      throw e
+    }
```

Comment:
I was reluctant to add these methods in the client because I didn't want to couple it to the `@globalfishingwatch/api-types` lib. However I understand that they will be for sure reused in other apps consuming this client. So I think we could leave it here unless you find a better place for them. WDYT?

Reply:
As we need the `fetchGuestUser` method to return the user in this format:
```js
{ id: 0, type: GUEST_USER_TYPE, permissions, groups: [] }
```
The only way I think we could remove the dependency from the `api-types` is adding a generic to the fetchGuestUser<User = unknown> and consume like:
```js
client.fetchGuestUser<UserData>()
```
where needed, for example  [here](https://github.com/GlobalFishingWatch/frontend/blob/feature/api-auth-with-next-review/apps/fishing-map/pages/index.tsx#L28) WDYT ?